codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerCodegenVisitor.kt

```diff
@@ -217,6 +217,7 @@ class ServerCodegenVisitor(context: PluginContext, private val codegenDecorator:
             rustCrate,
             protocolGenerator,
             protocolGeneratorFactory.support(),
+            protocolGeneratorFactory.protocol(codegenContext).httpBindingResolver,
```

Comment:
This passing this to the ServiceGenerator allows to use the instance of the `httpBindingResolver` specific to the current protocol everywhere. 

Without this the `OperationRegistryGenerator` was protocol specific.